services/director-v2/tests/unit/test_modules_dynamic_sidecar_client_api_public.py

```diff
@@ -0,0 +1,477 @@
+# pylint:disable=unused-argument
+# pylint:disable=redefined-outer-name
+
+from contextlib import contextmanager
+from typing import Any, AsyncIterable, Callable, Iterator, Optional, Type
+from unittest.mock import AsyncMock
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+from _pytest.monkeypatch import MonkeyPatch
+from fastapi import FastAPI, status
+from httpx import HTTPError, Response
+from pydantic import AnyHttpUrl, parse_obj_as
+from pytest_mock import MockerFixture
+from simcore_service_director_v2.core.settings import AppSettings
+from simcore_service_director_v2.modules.dynamic_sidecar.api_client._errors import (
+    ClientHttpError,
+    UnexpectedStatusError,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.api_client._public import (
+    DynamicSidecarClient,
+    get_dynamic_sidecar_client,
+    setup,
+    shutdown,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.errors import (
+    EntrypointContainerNotFoundError,
+    NodeportsDidNotFindNodeError,
+)
+
+# FIXTURES
+
+
+@pytest.fixture
+def dynamic_sidecar_endpoint() -> AnyHttpUrl:
+    return parse_obj_as(AnyHttpUrl, "http://missing-host:1111")
+
+
+@pytest.fixture
+def env_mocks(monkeypatch: MonkeyPatch, mock_env: None) -> None:
```

Comment:
```suggestion
def mock_end(monkeypatch: MonkeyPatch, mock_env: None) -> None:
```

Let's define this as good practice now. overwrite it, let's not have thousand different names

Reply:
OK, not bad, I like this one!

Reply:
actually mock_env not mock_end...

Reply:
So I can overwrite the same exiting mock with one of the same name even when inheriting from it!